src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -27,7 +27,7 @@ internal class MessagePump : IServer
 
         private volatile int _stopping;
         private int _outstandingRequests;
-        private readonly TaskCompletionSource<object> _shutdownSignal = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+        private readonly TaskCompletionSource<object?> _shutdownSignal = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
The result doesn't matter, we can use the new non-generic version.
```suggestion
        private readonly TaskCompletionSource _shutdownSignal = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```